Wrappers/Python/cil/framework/processors.py

```diff
@@ -0,0 +1,294 @@
+        else:
+            raise ValueError("Expected input dimensions is 2 or 3, got {0}"\
+                             .format(dataset.number_of_dimensions))
+    
+###### Example of DataProcessors
+
+class AX(DataProcessor):
+    '''Example DataProcessor
+    The AXPY routines perform a vector multiplication operation defined as
+
+    y := a*x
+    where:
+
+    a is a scalar
+
+    x a DataContainer.
+    '''
+    
+    def __init__(self):
+        kwargs = {'scalar':None, 
+                  'input':None, 
+                  }
+        
+        #DataProcessor.__init__(self, **kwargs)
+        super(AX, self).__init__(**kwargs)
+    
+    def check_input(self, dataset):
+        return True
+        
+    def process(self, out=None):
+        
+        dsi = self.get_input()
+        a = self.scalar
+        if out is None:
+            y = DataContainer(a * dsi.as_array(), True,
+                              dimension_labels=dsi.dimension_labels)
+            #self.setParameter(output_dataset=y)
+            return y
+        else:
+            out.fill(a * dsi.as_array())
+    
+
+###### Example of DataProcessors
+
+class CastDataContainer(DataProcessor):
+    '''Example DataProcessor
+    Cast a DataContainer array to a different type.
+
+    y := a*x
+    where:
+
+    a is a scalar
+
+    x a DataContainer.
+    '''
+    
+    def __init__(self, dtype=None):
+        kwargs = {'dtype':dtype, 
+                  'input':None, 
+                  }
+        
+        #DataProcessor.__init__(self, **kwargs)
+        super(CastDataContainer, self).__init__(**kwargs)
+    
+    def check_input(self, dataset):
+        return True
+        
+    def process(self, out=None):
+        
+        dsi = self.get_input()
+        dtype = self.dtype
+        if out is None:
+            y = numpy.asarray(dsi.as_array(), dtype=dtype)
+            
+            return type(dsi)(numpy.asarray(dsi.as_array(), dtype=dtype),
+                                dimension_labels=dsi.dimension_labels )
+        else:
+            out.fill(numpy.asarray(dsi.as_array(), dtype=dtype))
+    
+class PixelByPixelDataProcessor(DataProcessor):
+    '''Example DataProcessor
+    
+    This processor applies a python function to each pixel of the DataContainer
+    
+    f is a python function
+
+    x a DataSet.
+    '''
+    
+    def __init__(self):
+        kwargs = {'pyfunc':None, 
+                  'input':None, 
+                  }
+        #DataProcessor.__init__(self, **kwargs)
+        super(PixelByPixelDataProcessor, self).__init__(**kwargs)
+        
+    def check_input(self, dataset):
+        return True
+    
+    def process(self, out=None):
+        
+        pyfunc = self.pyfunc
+        dsi = self.get_input()
+        
+        eval_func = numpy.frompyfunc(pyfunc,1,1)
+
+        
+        y = DataContainer(eval_func(dsi.as_array()), True,
+                          dimension_labels=dsi.dimension_labels)
+        return y
```

Comment:
These are all examples and could be removed, possibly deprecated first

Reply:
moved to #1904